### PR TITLE
[onert/cl_common] Introduce cl-common library

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit introduces cl-common backend library.
It will be used by gpu-cl backend and acl-common library.

Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: #8786
Draft: #8808

### CODE REVIEW

**Comment by glistening** [comment]
@hseok-oh `cl-common` seems to be for almost everything.  Maybe exceptions are `cpu`, `ruy`, `xnnpack`?

**Comment by hseok-oh** [comment]
>Could you check it is right?

Right.

**Review by chunseoklee — DISMISSED** [review]

File: runtime/onert/backend/cl_common/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+file(GLOB SOURCES "*.cc")
```
Comment (glistening) [inline runtime/onert/backend/cl_common/CMakeLists.txt:1]: 'cl-common' seems header-only library. If so, 'SOURCES' looks redundant this line and the next lines including seems at this momenthis moment. Am I missing something?

Comment (hseok-oh) [reply runtime/onert/backend/cl_common/CMakeLists.txt:1]: I'll fix it.

**Comment by hseok-oh** [comment]
I've updated. PTAL

**Review by glistening — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**